src/store/student-store.js

```diff
@@ -18,13 +18,8 @@ class StudentStore extends Store{
 const store = new StudentStore();
 
 store.registerHandler('STUDENT_FOUND', data => {
-    student = {
-        //NOTE: this data is tentative
-        id : data.id,
-        name: data.name,
-        items: data.items,
-        hasOverdueItem: store.hasOverdueItems(data.items)
-    };
+    student = data;
+    student.hasOverdueItem = store.hasOverdueItems(data.items);
```

Comment:
This is not a good idea. `student.hasOverdueItem` might get set to `false` now, but ten minutes from now it will still be `false` even if it should be `true`. `hasOverdueItems` relies on the current time.